index.js

```diff
@@ -305,16 +243,20 @@ module.exports = {
       return prefixes;
     };
 
-    // Should return true if the user is an editor and thus
+    self.userIsEditor = function(req) {
+      return req.user;
+    };
+
+    // Should return true if the user is an editor and explicitly ask for viewing all
+    // by passing a showAll query parameter and thus
     // should bypass the normal restrictions on whether they
     // can see widgets and pieces for other personas, for
     // editing purposes. If this definition ("anyone who is
     // logged in is a potential editor") is not fine-grained
     // enough for your purposes, override this method at
     // project level
-
-    self.userIsEditor = function(req) {
-      return req.user;
+    self.userIsEditorAndWantToSeeAll = function(req) {
+      return self.userIsEditor(req) && req.query.showAll;
```

Comment:
This is going to be way too confusing for new users of the module and editors in general, no? It is not obvious why their content is gone if they don't add a special flag. A bigger concern is that if it isn't rendered, then it won't be saved - it will get erased from the database as soon as they touch the area editor. I don't think this property is a good idea unless it is flipped around the other way, as an explicit preview option, with a suitable UI and provision to block all area editing until you get out of preview mode so your content is safe.

Reply:
Actually, authors (and more) thought that persona did not work properly when in authoring, just because they did not see the impact of choosing a specific persona context to display a block.  
It was even creating non visible problems (end users did not the a block because it was choosen on author side to be persona specific - and authors did not understand the problem because they did see the block while authoring.  
And i am not sure to understand the the point about the fact that a non rendered block is not saved. I was able to select a non actual persona for a specific area - and then when switching from persona to persona, was able to see it only the the right context.  
Maybe i missed something on that?

Reply:
1. Create content with widgets of various personas in an area.
2. Switch to a specific persona. You will only presumably see widgets for that persona now on the page.
3. Make some edits to the area.
4. The widgets for other personas are gone.

Reply:
Fixed - as discussed with you, now the module displays areas even in bad context for author - just making them a bit translucent to indicates that this content will not be displayed for end users in this context.